iati/validator.py

```diff
@@ -445,6 +453,41 @@ def _parse_lxml_log_entry(log_entry):
     return error
 
 
+def _parse_ruleset_fail(rule):
+    """Parse a Rule skip or failure and convert it to a IATI ValidationError.
+
+    Args:
+        rule (iati.core.rulesets.Rule): The Rule which has either skipped or failed.
+
+    Returns:
+        ValidationError: An IATI ValidationError that contains the information from the log entry.
+
+    """
+    # undertake the mapping between Rule subclass and error name formats
+    rule_to_iati_error_mapping = {
+        'RuleAtLeastOne': 'err-rule-at-least-one-conformance-fail',
+        'RuleDateOrder': 'err-rule-date-order-conformance-fail',
+        'RuleDependent': 'err-rule-dependent-conformance-fail',
+        'RuleNoMoreThanOne': 'err-rule-no-more-than-one-conformance-fail',
+        'RuleRegexMatches': 'err-rule-regex-matches-conformance-fail',
+        'RuleRegexNoMatches': 'err-rule-regex-no-matches-conformance-fail',
+        'RuleStartsWith': 'err-rule-starts-with-conformance-fail',
+        'RuleSum': 'err-rule-sum-conformance-fail',
+        'RuleUnique': 'err-rule-unique-conformance-fail'
+    }
+
+    rule_name = rule.__class__.__name__
```

Comment:
Rules have a `name` property which can use used directly rather than having to work with the class names.